algebra/ring.lean

```diff
@@ -15,6 +15,85 @@ section
   (left_distrib n 1 1).trans (by simp)
 end
 
+section units
+  variables [semiring α]
+  variable (α)
+
+  structure invertible :=
+    (val : α)
+    (inv : α)
+    (property : val * inv = 1 ∧ inv * val = 1)
```

Comment:
* style: don't indent the properties.
* Don't use conjunctions in a structure if two fields will do. Split `property` into `val_inv` and `inv_val`.
* Call the structure `units A` instead of `invertible A`.